practice/validator/site.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+/**
+ * Tests whether the string matches the regular expression
+ * @param  {string} testString The string to test
+ * @param  {reg exp} regExp    The regular expression.  Not a string.
+ * @return {boolean}           True if there is a match, false otherwise.
+ */
+function stringValidator(testString, regExp) {
+  return regExp.test(testString);
+}
+
+/**
+ * The the string entered on the form.
+ * @param  {jQuery selector} entryItem The selector for a text input.
+ * @return {string}                    The string entered on the form.
+ */
+function getFieldEntry(entryItem) {
+  return entryItem.val();
+}
+
+/**
+ * displayInvalid() highlights the input box and un-hides the warning message.
+ */
+function displayInvalid(entryItem) {
+  entryItem.addClass('invalid');
+  $(entryItem).next().addClass('visible-warning');
```

Comment:
If I understand this correctly, this chooses the item following the entryItem, and adds the visible-warning class to it. I'm curious why you chose this route rather than a class name or the like?